### PR TITLE
Video Player 📼

### PR DESCRIPTION
**Pull request by namniak**
Includes:
- `VideoPlayer` + controls (supports subtitles, mute, fs). Is accessible. Works as both, inline player and background video. Has a pretty goog API for customization
- `MobileFullscreensplayer` - for fullscreen/native player depending on device
- Brought latest version for existing packages

### CODE REVIEW

File: templates/react/src/components/MobileFsVideo/index.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    if (document.fullscreen !== undefined) {
+      return document.fullscreen;
+    }
+    if (document.mozFullScreen !== undefined) {
+      return document.mozFullScreen;
+    }
+    if (document.webkitIsFullScreen !== undefined) {
+      return document.webkitIsFullScreen;
+    }
+  };
```
Comment (jeremenichelli) [inline templates/react/src/components/MobileFsVideo/index.js]: Wouldn't it make sense to have these three functions inside the util so they are available to any component and we don't encourage code repetition?

Comment (namniak) [inline templates/react/src/components/MobileFsVideo/index.js]: Maybe yes. 
in fact, my original intention was to publish this component that's why it's self-contained. Still want to do it maybe next week but I hate writing the readme... 
It doesn't really need to be in the generator.

**Comment by namniak** [comment]
@njam3 renamed

File: templates/base/src/util/seconds-to-minutes.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  if (minutes < 10) {
+    minutes = `0${minutes}`;
+  }
+
+  if (seconds < 10) {
+    seconds = `0${seconds}`;
+  }
+
+  return `${minutes}:${seconds}`;
+}
```
Comment (iranreyes) [inline templates/base/src/util/seconds-to-minutes.js]: This is nice, we should start to include unit tests mainly for the utils. They are easily tested and ensure we cover all the cases.

In the case of this function, will be cool if we protect the `totalSeconds` argument against NaN values and negative values.

File: templates/react/src/components/MobileFullscreenVideo/index.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import fullScreen from 'fullscreen-handler';
+
+const isIOS = Boolean(typeof navigator !== 'undefined' && navigator.appVersion.match(/OS (\d+)_(\d+)_?(\d+)?/));
```
Comment (iranreyes) [inline templates/react/src/components/MobileFullscreenVideo/index.js]: This constant is the same than `detect.isIOS`

File: templates/react/src/components/MobileFullscreenVideo/index.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      top: window.innerHeight / 2,
+      left: window.innerWidth / 2,
+    }, this.props.style);
+
+    const className = 'MobileFullscreenVideo';
+
+    return (
+      <video
+        style={style}
+        className={`${className} ${this.props.className}`}
```
Comment (iranreyes) [inline templates/react/src/components/MobileFullscreenVideo/index.js]: We should consider using something like `classnames`

File: templates/react/src/components/MobileFullscreenVideo/index.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  render() {
+    const style = Object.assign({}, {
+      position: 'fixed',
+      width: 0,
+      height: 0,
+      top: window.innerHeight / 2,
+      left: window.innerWidth / 2,
+    }, this.props.style);
+
+    const className = 'MobileFullscreenVideo';
```
Comment (iranreyes) [inline templates/react/src/components/MobileFullscreenVideo/index.js]: Constants should not be inside the render :-)

File: templates/react/src/test/MobileFullscreenVideo/index.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  handleOnClick = () => {
+    this.video.play();
+  };
+
+  render() {
+    const props = this.props;
+
+    return (
+      <div
```
Comment (iranreyes) [inline templates/react/src/test/MobileFullscreenVideo/index.js]: If the styles are not dynamic they should go to the CSS given is more work the render at least the idea is to have them inline

Comment (namniak) [reply templates/react/src/test/MobileFullscreenVideo/index.js]: this is tests

File: templates/react/src/components/MobileFullscreenVideo/index.js (hunk 3 above)
Comment (namniak) [reply templates/react/src/components/MobileFullscreenVideo/index.js]: changed

File: templates/react/src/components/MobileFullscreenVideo/index.js (hunk 5 above)
Comment (namniak) [reply templates/react/src/components/MobileFullscreenVideo/index.js]: fixed

File: templates/react/src/components/MobileFullscreenVideo/index.js (hunk 4 above)
Comment (namniak) [reply templates/react/src/components/MobileFullscreenVideo/index.js]: yeah. let's merge existing PRs esp 'connect' one and we'll improve

**Pull request merged**